### PR TITLE
adds new send-text option, comments, and readme example

### PR DESCRIPTION
**Pull request by miloh**
enables bannery style text scrolling
- VerticalDrawText() scrolls text from bottom to top
- use CharacterWidth of 'W' from bdf-font for font width, does not assume
  flaschen taschen dimensions

### CODE REVIEW

**Comment by miloh** [comment]
problem with getting this stanza to work:

```
./send-text -h localhost "♥Flaschen" -f fonts/5x5.bdf -s 0 -g 45x35+0+15 & ./send-text -h localhost "Tasche" -f fonts/5x5.bdf  -v  -s 0 -g 45x35+20+0
```

File: client/bdf-font.cc [hunk 1]
```diff
@@ -150,4 +150,16 @@ int DrawText(FlaschenTaschen *c, const Font &font,
     return x - start_x;
 }
 
+int VerticalDrawText(FlaschenTaschen *c, const Font &font,
+             int x, int y, const Color &color, const Color *background_color,
```
Comment (hzeller) [inline client/bdf-font.cc:154]: The indentation looks off here; the parameters should align with the opening parenthesis above.
(sorry, I am a bit picky with code :) )

File: client/send-text.cc [hunk 2]
```diff
@@ -26,6 +26,10 @@
 
 #include <string>
 
+#define DEFAULT_WIDTH 45
+#define DEFAULT_HEIGHT 35
+#define WIDEST_GLYPH 'Z'
```
Comment (hzeller) [inline client/send-text.cc]: Maybe 'W' would be a good wide glyph.

**Comment by miloh** [comment]
did you get to test it on the flaschen taschen?

File: client/send-text.cc [hunk 3]
```diff
@@ -158,28 +172,53 @@ int main(int argc, char *argv[]) {
 
     // Center in in the available display space.
     const int y_pos = (height - font.height()) / 2 + font.baseline();
+    const int x_pos = (width - font.CharacterWidth(WIDEST_GLYPH)) / 2 ;
 
     // dry-run to determine total number of pixels.
-    const int total_len = DrawText(&display, font, 0, y_pos, fg, NULL, text);
+       const int total_height = DrawText(&display, font, 0, y_pos, fg, NULL, text);
```
Comment (hzeller) [inline client/send-text.cc]: Why is this indented more than the previous lines ?

Comment (miloh) [inline client/send-text.cc]: fixed

**Comment by miloh** [comment]
now this is giving me a problem.

expected behavior?

```
./send-text -h localhost "♥Flaschen" -f fonts/5x5.bdf -s 30 -b ff00ff -g 45x35+0+15+2 & ./send-text -h localhost "Taschen" -f fonts/5x5.bdf  -v  -s 30 -g 45x35+20+0+3 -b 00ff00 && fg
```

**Comment by hzeller** [comment]
So it looks good, it does not flash when it is on different layers:

https://youtu.be/ys7jlcsApho

The vertical looks like it is squished together, as if it is not using 5
pixels distance

**Comment by miloh** [comment]
freeze the frame -- it fits.  perhaps it is best viewed statically with -s 0

**Comment by hzeller** [comment]
On 8 May 2016 at 23:40, miloh notifications@github.com wrote:

> freeze the frame -- it fits. perhaps it is best viewed statically with -s 0

Yep, good idea. I am at home already (need to get up before 6am tomorrow),
but something to test.

-h

**Comment by hzeller** [comment]
Thanks! merged.

**Pull request closed without merging**